### PR TITLE
[Notifier] [Nexmo] Add tests

### PR DESCRIPTION
**Pull request by OskarStark**
| Q             | A
| ------------- | ---
| Branch?       | 5.1
| Bug fix?      | yes
| New feature?  | no 
| Deprecations? | no 
| Tickets       | ---
| License       | MIT
| Doc PR        | ---

### CODE REVIEW

File: src/Symfony/Component/Notifier/Bridge/Nexmo/NexmoTransportFactory.php [hunk 1]
```diff
@@ -32,6 +33,11 @@ public function create(Dsn $dsn): TransportInterface
         $apiKey = $this->getUser($dsn);
         $apiSecret = $this->getPassword($dsn);
         $from = $dsn->getOption('from');
+
+        if (!$from) {
+            throw new IncompleteDsnException('Missing from.', $dsn->getOriginalDsn());
+        }
```
Comment (OskarStark) [inline src/Symfony/Component/Notifier/Bridge/Nexmo/NexmoTransportFactory.php]: This should be considered a bugfix

**Comment by OskarStark** [comment]
Travis failure looks unrelated

File: src/Symfony/Component/Notifier/Bridge/Nexmo/NexmoTransportFactory.php (hunk 1 above)
Comment (derrabus) [reply src/Symfony/Component/Notifier/Bridge/Nexmo/NexmoTransportFactory.php]: see https://github.com/symfony/symfony/pull/39406#discussion_r539307754

**Comment by OskarStark** [comment]
Travis failure looks unrelated and is only in 7.4 job:
```
There were 7 errors:
1) Symfony\Component\Notifier\Bridge\Nexmo\Tests\NexmoTransportFactoryTest::testCreateWithDsn
Error: Class 'Symfony\Component\EventDispatcher\LegacyEventDispatcherProxy' not found
/home/travis/build/symfony/symfony/src/Symfony/Component/Notifier/Bridge/Nexmo/vendor/symfony/notifier/Transport/AbstractTransportFactory.php:33
/home/travis/build/symfony/symfony/src/Symfony/Component/Notifier/Bridge/Nexmo/Tests/NexmoTransportFactoryTest.php:66
/home/travis/build/symfony/symfony/src/Symfony/Component/Notifier/Bridge/Nexmo/Tests/NexmoTransportFactoryTest.php:24
2) Symfony\Component\Notifier\Bridge\Nexmo\Tests\NexmoTransportFactoryTest::testCreateWithNoFromThrowsMalformed
Error: Class 'Symfony\Component\EventDispatcher\LegacyEventDispatcherProxy' not found
/home/travis/build/symfony/symfony/src/Symfony/Component/Notifier/Bridge/Nexmo/vendor/symfony/notifier/Transport/AbstractTransportFactory.php:33
/home/travis/build/symfony/symfony/src/Symfony/Component/Notifier/Bridge/Nexmo/Tests/NexmoTransportFactoryTest.php:66
/home/travis/build/symfony/symfony/src/Symfony/Component/Notifier/Bridge/Nexmo/Tests/NexmoTransportFactoryTest.php:35
3) Symfony\Component\Notifier\Bridge\Nexmo\Tests\NexmoTransportFactoryTest::testSupportsNexmoScheme
Error: Class 'Symfony\Component\EventDispatcher\LegacyEventDispatcherProxy' not found
/home/travis/build/symfony/symfony/src/Symfony/Component/Notifier/Bridge/Nexmo/vendor/symfony/notifier/Transport/AbstractTransportFactory.php:33
/home/travis/build/symfony/symfony/src/Symfony/Component/Notifier/Bridge/Nexmo/Tests/NexmoTransportFactoryTest.php:66
/home/travis/build/symfony/symfony/src/Symfony/Component/Notifier/Bridge/Nexmo/Tests/NexmoTransportFactoryTest.php:45
4) Symfony\Component\Notifier\Bridge\Nexmo\Tests\NexmoTransportFactoryTest::testNonNexmoSchemeThrows
Error: Class 'Symfony\Component\EventDispatcher\LegacyEventDispatcherProxy' not found
/home/travis/build/symfony/symfony/src/Symfony/Component/Notifier/Bridge/Nexmo/vendor/symfony/notifier/Transport/AbstractTransportFactory.php:33
/home/travis/build/symfony/symfony/src/Symfony/Component/Notifier/Bridge/Nexmo/Tests/NexmoTransportFactoryTest.php:66
/home/travis/build/symfony/symfony/src/Symfony/Component/Notifier/Bridge/Nexmo/Tests/NexmoTransportFactoryTest.php:56
5) Symfony\Component\Notifier\Bridge\Nexmo\Tests\NexmoTransportTest::testToStringContainsProperties
Error: Class 'Symfony\Component\EventDispatcher\LegacyEventDispatcherProxy' not found
/home/travis/build/symfony/symfony/src/Symfony/Component/Notifier/Bridge/Nexmo/vendor/symfony/notifier/Transport/AbstractTransport.php:48
/home/travis/build/symfony/symfony/src/Symfony/Component/Notifier/Bridge/Nexmo/NexmoTransport.php:41
/home/travis/build/symfony/symfony/src/Symfony/Component/Notifier/Bridge/Nexmo/Tests/NexmoTransportTest.php:49
/home/travis/build/symfony/symfony/src/Symfony/Component/Notifier/Bridge/Nexmo/Tests/NexmoTransportTest.php:25
6) Symfony\Component\Notifier\Bridge\Nexmo\Tests\NexmoTransportTest::testSupportsMessageInterface
Error: Class 'Symfony\Component\EventDispatcher\LegacyEventDispatcherProxy' not found
/home/travis/build/symfony/symfony/src/Symfony/Component/Notifier/Bridge/Nexmo/vendor/symfony/notifier/Transport/AbstractTransport.php:48
/home/travis/build/symfony/symfony/src/Symfony/Component/Notifier/Bridge/Nexmo/NexmoTransport.php:41
/home/travis/build/symfony/symfony/src/Symfony/Component/Notifier/Bridge/Nexmo/Tests/NexmoTransportTest.php:49
/home/travis/build/symfony/symfony/src/Symfony/Component/Notifier/Bridge/Nexmo/Tests/NexmoTransportTest.php:32
7) Symfony\Component\Notifier\Bridge\Nexmo\Tests\NexmoTransportTest::testSendNonSmsMessageThrowsException
Error: Class 'Symfony\Component\EventDispatcher\LegacyEventDispatcherProxy' not found
/home/travis/build/symfony/symfony/src/Symfony/Component/Notifier/Bridge/Nexmo/vendor/symfony/notifier/Transport/AbstractTransport.php:48
/home/travis/build/symfony/symfony/src/Symfony/Component/Notifier/Bridge/Nexmo/NexmoTransport.php:41
/home/travis/build/symfony/symfony/src/Symfony/Component/Notifier/Bridge/Nexmo/Tests/NexmoTransportTest.php:49
/home/travis/build/symfony/symfony/src/Symfony/Component/Notifier/Bridge/Nexmo/Tests/NexmoTransportTest.php:40
```

**Comment by jderusse** [comment]
That mean the class `Symfony\Component\EventDispatcher\Event` exists but not `Symfony\Component\EventDispatcher\LegacyEventDispatcherProxy`.

I believe your should add a conflict rule in composer.json (siilar to Messenger component)
```
    "conflict": {
        "symfony/event-dispatcher": "<4.4",
    }
```

**Comment by derrabus** [comment]
No, this is a bug in Notifier 5.0.0 that is used here (because we're testing with lowest deps here). We should bump `symfony/notifier` to `^5.1` in the nexmo bridge (and possibly in others too).

**Comment by OskarStark** [comment]
> We should bump symfony/notifier to ^5.1 in the nexmo bridge (and possibly in others too).

Done

File: src/Symfony/Component/Notifier/Bridge/Nexmo/composer.json [hunk 2]
```diff
@@ -18,7 +18,7 @@
     "require": {
         "php": ">=7.2.5",
         "symfony/http-client": "^4.3|^5.0",
-        "symfony/notifier": "^5.0,<5.2"
+        "symfony/notifier": "^5.1,<5.2"
```
Comment (derrabus) [inline src/Symfony/Component/Notifier/Bridge/Nexmo/composer.json]:
```suggestion
        "symfony/notifier": "~5.1.0"
```

🤓

Comment (OskarStark) [reply src/Symfony/Component/Notifier/Bridge/Nexmo/composer.json]: ok

Comment (OskarStark) [inline src/Symfony/Component/Notifier/Bridge/Nexmo/composer.json]:
```suggestion
        "symfony/notifier": "~5.1.0"
```

**Review by derrabus — APPROVED** [review]

**Comment by nicolas-grekas** [comment]
Thank you @OskarStark.

**Pull request merged**